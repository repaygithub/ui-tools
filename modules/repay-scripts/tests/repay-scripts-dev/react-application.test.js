const TestSetup = require('../test-setup')

beforeEach(() => jest.setTimeout(1000 * 60 * 5))

describe('@repay/repay-scripts', () => {
  it('should host a react application for local development', async () => {
    const test = new TestSetup('react-application')
    await test.setup()
    test.spawn('yarn', [
      'repay-scripts',
      'dev',
      '--babel-env',
      'production',
      'index.js',
      '--port',
      '9696',
    ])

    await test.waitForText('Compiled successfully')
    const page = await test.getPage()
    await page.goto('https://localhost:9696')
    const content = await page.$eval('.react-root', el => el.parentElement.outerHTML)
    expect(content).toMatchSnapshot()
  })
})
