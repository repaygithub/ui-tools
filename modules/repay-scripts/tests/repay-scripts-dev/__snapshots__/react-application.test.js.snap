// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@repay/repay-scripts should host a react application for local development 1`] = `"<div class=\\"app-root\\"><div class=\\"react-root\\">Testing Text</div></div>"`;
